packages/strategy-tests/src/lib.rs

```diff
@@ -1120,17 +1118,111 @@ impl Strategy {
+                                        let name = doc_type_clone.remove("title").expect(
+                                            "Expected to get a doc type title in ContractCreate",
+                                        );
+                                        if let Value::Map(ref mut map) = doc_types {
+                                            map.push((
+                                                Value::Text(name.to_string()),
+                                                doc_type_clone,
+                                            ));
+                                        } else {
+                                            error!("doc_types is not a Value::Map as expected");
+                                        }
+                                    }
+                                    Err(e) => {
+                                        error!("Error generating random document type: {:?}", e)
+                                    }
+                                }
+                            }
+
+                            let contract_factory = match DataContractFactory::new(
+                                platform_version.protocol_version,
+                            ) {
+                                Ok(contract_factory) => contract_factory,
+                                Err(e) => {
+                                    error!("Failed to get DataContractFactory while creating random contract: {e}");
+                                    continue;
+                                }
+                            };
+
+                            let created_data_contract = match contract_factory.create(
+                                owner_id,
+                                *identity_nonce,
+                                doc_types,
+                                None,
+                                None,
+                            ) {
+                                Ok(contract) => contract,
+                                Err(e) => {
+                                    error!("Failed to create random data contract: {e}");
+                                    continue;
+                                }
+                            };
+
+                            let transition = match contract_factory
+                                .create_data_contract_create_transition(created_data_contract)
+                            {
+                                Ok(transition) => transition,
+                                Err(e) => {
+                                    error!("Failed to create ContractCreate transition: {e}");
+                                    continue;
+                                }
+                            };
+
+                            // Sign transition
+                            let public_key = identity
+                                .get_public_key_by_id(2 as KeyID)
```

Comment:
you shouldn't rely on id 2. Just ask for an existing authentication key with the level needed for the state transition. (I think it's critical)